scenario_player/runner.py

```diff
@@ -2,46 +2,259 @@
-from scenario_player.utils import TimeOutHTTPAdapter, get_udc_and_token, wait_for_txs
-from scenario_player.utils.token import Token, UserDepositContract
+from scenario_player.utils import TimeOutHTTPAdapter
+from scenario_player.utils.configuration.settings import SettingsConfig, UDCSettingsConfig
+from scenario_player.utils.token import (
+    TokenDetails,
+    eth_maybe_transfer,
+    load_token_configuration_from_file,
+    save_token_configuration_to_file,
+    token_maybe_mint,
+    userdeposit_maybe_deposit,
+    userdeposit_maybe_increase_allowance,
+)
 
 if TYPE_CHECKING:
     from scenario_player.tasks.base import Task, TaskState
 
 log = structlog.get_logger(__name__)
 
 
+# The `mint` function checks for overflow of the total supply. Here we
+# have a large enough number for both scenario runs and the scenario
+# orchestration account capacity.
+NUMBER_OF_RUNS_BEFORE_OVERFLOW = 2 ** 64
+ORCHESTRATION_MAXIMUM_BALANCE = UINT256_MAX // NUMBER_OF_RUNS_BEFORE_OVERFLOW
+
+
+def is_udc_enabled(udc_settings: UDCSettingsConfig):
+    should_deposit_ud_token = udc_settings.token.deposit
+    return udc_settings.enable and should_deposit_ud_token
+
+
+def wait_for_nodes_to_be_ready(node_runners, session):
+    with gevent.Timeout(MAX_RAIDEN_STARTUP_TIME):
+        for node_runner in node_runners:
+            url = f"http://{node_runner.base_url}/api/v1/address"
+            while True:
+                try:
+                    if session.get(url).ok:
+                        break
+                except requests.exceptions.RequestException:
+                    pass
+                gevent.sleep(0.5)
+
+
+def get_udc_and_corresponding_token_from_dependencies(
+    udc_settings: UDCSettingsConfig, settings: SettingsConfig, proxy_manager: ProxyManager
+) -> Tuple[UserDeposit, CustomToken]:
+    """ Return contract proxies for the UserDepositContract and associated token.
+
+    This will return a proxy to the `UserDeposit` contract as determined by the
+    **local** Raiden depedency.
+    """
+    assert udc_settings.enable
+    chain_id = settings.chain_id
+    assert chain_id, "Missing configuration, either set udc_address or the chain_id"
+
+    udc_address = udc_settings.address
+
+    if udc_address is None:
+
+        contracts = get_contracts_deployment_info(chain_id, version=RAIDEN_CONTRACT_VERSION)
+
+        msg = (
+            f"invalid chain_id, {chain_id} is not available for version {RAIDEN_CONTRACT_VERSION}"
+        )
+        assert contracts, msg
+
+        udc_address = contracts["contracts"][CONTRACT_USER_DEPOSIT]["address"]
+
+    userdeposit_proxy = proxy_manager.user_deposit(
+        UserDepositAddress(to_canonical_address(udc_address)), "latest"
+    )
+
+    token_address = userdeposit_proxy.token_address("latest")
+    user_token_proxy = proxy_manager.custom_token(token_address, "latest")
+
+    return userdeposit_proxy, user_token_proxy
+
+
+def get_token_network_registry_from_dependencies(
```

Comment:
This sort of replaces this RPC call:

https://github.com/raiden-network/scenario-player/blob/edd041e816776c141a1182f089ab6bb75a976e91/scenario_player/runner.py#L257-L271

There is a problem here, since the scenario player version of the contracts may be different than the version used by the raiden client. This is also true for the user deposit and token smart contracts.

We decided that instead of having a mix of solutions that work sometimes and not other we have decided to implement the simple solution first, which is to just rely on manual synchronization of the dependencies. This should not be a big problem since package releases are spaced out.

A proper solution would require a new command in raiden to expose the node's configuration.